### PR TITLE
[featureflag] document instrumentation

### PR DESCRIPTION
**Pull request by julianocosta89**
Adds documentation for feature flag service, focused on instrumentation code.

### CODE REVIEW

File: docs/services/featureflagservice.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# featute flag service
```
Comment (austinlparker) [inline docs/services/featureflagservice.md:1]:
```suggestion
# Feature Flag Service
```

File: docs/services/featureflagservice.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+This service is written in Erlang/Elixir and it is responsible for creating,
+reading, updating and deleting feature flags in a PostgreSQL DB.
+It is called by Product Catalog and Shipping services.
+
+[featute flag service source](../../src/featureflagservice/)
```
Comment (austinlparker) [inline docs/services/featureflagservice.md:7]:
```suggestion
[Feature Flag Service Source](../../src/featureflagservice/)
```

File: docs/services/featureflagservice.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+### Initialize OpenTelemetry Phoenix instrumentation
+
+Traces of interaction with the web interface are provided by the OpenTelemetry
+[Phoenix
+instrumentation](https://github.com/open-telemetry/opentelemetry-erlang-contrib/tree/main/instrumentation/opentelemetry_phoenix)
+with Spans for database queries added through the [Ecto
+instrumentation](https://github.com/open-telemetry/opentelemetry-erlang-contrib/tree/main/instrumentation/opentelemetry_ecto).
+
+This is done in the `application.ex` as it follows:
```
Comment (austinlparker) [inline docs/services/featureflagservice.md:19]:
```suggestion
This is done in the `application.ex` as follows:
```

File: docs/services/featureflagservice.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+### Add gRPC instrumentation
+
+The GRPC service uses [grpcbox](https://github.com/tsloughter/grpcbox) and uses
+the [grpcbox
+interceptor](https://github.com/open-telemetry/opentelemetry-erlang-contrib/tree/main/instrumentation/opentelemetry_grpcbox)
+for instrumentation.
+
+This is configured in the `runtime.exs` file, as it follows:
```
Comment (austinlparker) [inline docs/services/featureflagservice.md:52]:
```suggestion
This is configured in the `runtime.exs` file, as follows:
```

**Review by austinlparker — APPROVED** [review]
Couple of nits that'd be good to fix but otherwise lgtm.

File: docs/services/featureflagservice.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        :services => %{:"hipstershop.FeatureFlagService" => :ffs_service}
+      },
+      :listen_opts => %{:port => grpc_port}
+    }
+  ]
+```
+
+### Add attributes to auto-instrumented spans
+
+Adding attributes to a span is accomplished using `?set_attribute` on the span
```
Comment (fatsheep9146) [inline docs/services/featureflagservice.md:70]: is accomplished using -> is accomplished by using

File: docs/services/featureflagservice.md (hunk 1 above)
Comment (julianocosta89) [reply docs/services/featureflagservice.md:1]: All service docs are lower case.
It also bothers me, but I have just followed the pattern.

Comment (mviitane) [reply docs/services/featureflagservice.md:1]: We seem to have inconsistencies about the capital letters in service names throughout the docs. Maybe this should be fixed with another PR in all places. Anyway, I'd also start with a capital letter.

@julianocosta89 note also the typo.

**Review by puckpuck — APPROVED** [review]

**Review by mviitane — APPROVED** [review]
LGTM

**Pull request merged**